File: ios/GuardianApp/Models/Threat.swift
import Foundation

struct Threat: Codable, Identifiable {
    let id: String
    let severity: Int // 0-100
    let title: String
    let description: String
    let timestamp: Date
    let account: String
    let service: String
    let evidence: [String]

    enum CodingKeys: String, CodingKey {
        case id
        case severity
        case title
        case description
        case timestamp
        case account
        case service
        case evidence
    }
}

struct ThreatTimeline: Codable {
    let threats: [Threat]
    let total: Int
}

struct ThreatStats: Codable {
    let criticalCount: Int
    let highCount: Int
    let mediumCount: Int
    let lowCount: Int
}

File: android/app/src/main/kotlin/com/aws/guardian/models/Threat.kt
package com.aws.guardian.models

import com.google.firebase.database.IgnoreExtraProperties
import java.io.Serializable

@IgnoreExtraProperties
data class Threat(
    val id: String = "",
    val severity: Int = 0,  // 0-100
    val title: String = "",
    val description: String = "",
    val timestamp: Long = 0L,
    val account: String = "",
    val service: String = "",
    val evidence: List<String> = emptyList()
) : Serializable

data class ThreatTimeline(
    val threats: List<Threat> = emptyList(),
    val total: Int = 0
)

data class ThreatStats(
    val criticalCount: Int = 0,
    val highCount: Int = 0,
    val mediumCount: Int = 0,
    val lowCount: Int = 0
)
